celery/backends/dynamodb.py

```diff
@@ -194,6 +222,17 @@ def _get_or_create_table(self):
                     self.table_name
                 )
             )
+            # Enable time-to-live on the table, ignoring whether or not TTL is
+            # currently specified as an option for the backend. This allows
+            # enabling it later. Until then, items are inserted without a value
+            # in the ttl field, meaning that DynamoDB will never expire them.
+            self._client.update_time_to_live(**self._get_ttl_specification())
```

Comment:
According to boto3 [documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html?highlight=dynamodb#DynamoDB.Client.update_time_to_live):

> It can take up to one hour for the change to fully process. Any additional UpdateTimeToLive calls for the same table during this one hour duration result in a ValidationException .

So perhaps we need to catch the `ValidationException` and log it?

Reply:
This has been addressed, but perhaps not sufficiently:

The TTL handling is now improved so that trying to enable or disable it when it is already enabled/enabling or disabled/disabling, is successful.

Other cases are more challenging:
- TTL is already enabled, but on the wrong attribute: This requires disabling it and then re-enabling it, but the operation throttling prevents this from being done quickly.
- TTL is being enabled or disabled and we want to instead disable/enable it: The operation throttling prevents this from being done in quick succession.

In the challenging cases, the exception is simply raised. The exception itself has an informative `['Error']['Message']` field, but it will perhaps not be logged appropriately to the user?

Should I rather catch these and extract the error message, and log it explicitly with level 'error' or something?

Reply:
I think logging the error message and reraising the exception would be preferrable, what do you think?

Reply:
Agreed and addressed. Let me know if it should be done differently.